src/_pytest/terminal.py

```diff
@@ -483,43 +484,51 @@ def pytest_runtest_logreport(self, report: TestReport) -> None:
 
     @property
     def _is_last_item(self):
-        return len(self._progress_nodeids_reported) == self._session.testscollected
+        return self._progress_items_reported == self._session.testscollected
 
-    def pytest_runtest_logfinish(self, nodeid):
```

Comment:
Is the change from `pytest_runtest_logfinish` to `pytest_runtest_teardown` + `pytest_runtestloop` part of the fix or just a refactor?

Reply:
Part of the fix IIRC.